service/topology_coordinator.cc

```diff
@@ -803,6 +804,9 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
                         co_await coroutine::maybe_yield();
                     });
                 }
+                } else {
+                    error = "Can't ALTER keyspace " + ks_name + ", keyspace doesn't exist";
+                }
 
                 updates.push_back(canonical_mutation(topology_mutation_builder(guard.write_timestamp())
```

Comment:
Shouldn't we skip entering `tablet_migration` and increasing topology version if ALTER fails (there's an `error`)?

Reply:
I see that this problem is pre-existing, but I'd like to understand why is this done so nevertheless, and maybe it needs a separate issue.

Reply:
If preparing ALTER mutations (schema mutations or tablets mutations) fails, we still need to delete the global `rf_change` req along with the topo req tracking entry. 
So whether ALTER succeeds or not, we need to perform some mutations.

Reply:
This doesn't answer my question. Yes we need to perform **some** mutations, the ones you mentioned. But what about these things that I asked:
> entering tablet_migration and increasing topology version

?

Reply:
That needs to be investigated, I don't know the answer off the top of my head. Added to https://github.com/scylladb/scylladb/issues/19123